forc-plugins/forc-client/src/op/call/list_functions.rs

```diff
@@ -0,0 +1,161 @@
+    } else {
+        let type_lookup = unified_program_abi
+            .types
+            .iter()
+            .map(|decl| (decl.type_id, decl.clone()))
+            .collect::<HashMap<_, _>>();
+
+        for func in &unified_program_abi.functions {
+            let func_args = func
+                .inputs
+                .iter()
+                .map(|input| {
+                    let Ok(param_type) = ParamType::try_from_type_application(input, &type_lookup)
+                    else {
+                        return Err(anyhow!("Failed to convert input type application"));
+                    };
+                    let func_args =
+                        format!("{}: {}", input.name, param_to_function_arg(&param_type));
+                    let func_args_input = {
+                        let token =
+                            param_type_val_to_token(&param_type, &get_default_value(&param_type))
+                                .map_err(|err| {
+                                anyhow!(
+                                    "Failed to generate example call for {}: {}",
+                                    func.name,
+                                    err
+                                )
+                            })?;
+                        token_to_string(&token).map_err(|err| {
+                            anyhow!(
+                                "Failed to convert token to string for {}: {}",
+                                func.name,
+                                err
+                            )
+                        })?
+                    };
+                    Ok((func_args, func_args_input))
+                })
+                .collect::<Result<Vec<_>>>()?;
+
+            let func_args_types = func_args
+                .iter()
+                .map(|(func_args, _)| func_args.to_owned())
+                .collect::<Vec<String>>()
+                .join(", ");
+
+            let func_args_inputs = func_args
+                .iter()
+                .map(|(_, func_args_input)| format!("\"{}\"", func_args_input))
+                .collect::<Vec<String>>()
+                .join(" ");
+
+            let return_type = ParamType::try_from_type_application(&func.output, &type_lookup)
+                .map(|param_type| param_to_function_arg(&param_type))
+                .map_err(|err| {
+                    anyhow!(
+                        "Failed to convert output type application for {}: {}",
+                        func.name,
+                        err
+                    )
+                })?;
+
+            // Get the ABI path or URL as a string
+            let raw_abi_input = match abi {
+                Either::Left(path) => path.to_str().unwrap_or("").to_owned(),
+                Either::Right(url) => url.to_string(),
+            };
+
+            writeln!(
+                writer,
+                "{}({}) -> {}",
+                func.name, func_args_types, return_type
+            )?;
```

Comment:
Using color here makes it easeier to scan the function names:

```suggestion
            let painted_name = forc_util::ansiterm::Colour::Blue.paint(func.name.clone());

            writeln!(
                writer,
                "{}({}) -> {}",
                painted_name, func_args_types, return_type
            )?;
```

![image](https://github.com/user-attachments/assets/8f7662f8-afd5-4c73-b5e7-8c4d302e2ddc)

Reply:
https://github.com/FuelLabs/sway/pull/7018/commits/67561aa3a17a44ef705a609fa11f2e27d14bc3dc